airflow/utils/helpers.py

```diff
@@ -186,3 +158,117 @@ def f(t):
         s += pattern % tuple(f(t) for t in line) + '\n'
     s += separator + '\n'
     return s
+
+
+class AirflowImporter(object):
+    """
+    Importer that dynamically loads a class and module from its parent. This
+    allows Airflow to support `from airflow.operators.bash_operator import
+    BashOperator` even though BashOperator is actually in
+    airflow.operators.bash_operator.
+
+    The importer also takes over for the parent_module by wrapping it. This is
+    required to support attribute-based usage:
+
+        from airflow import operators
+        operators.BashOperator(...)
+    """
+
+    def __init__(self, parent_module, module_attributes):
+        """
+        :param parent_module: The string package name of the parent module. For
+            example, 'airflow.operators'
+        :type parent_module: string
+        :param module_attributes: The file to class mappings for all importable
+            classes.
+        :type module_attributes: string
+        """
+        self.parent_module = parent_module
```

Comment:
Suggestion: make these three private (`self._parent_module, self._attribute_modules, self._loaded_modules`). With a fresh `import airflow`, they look like the only attributes in the `operators` module using Python autocomplete
```
In [2]: airflow.operators.<tab>
airflow.operators.attribute_modules  airflow.operators.loaded_modules     airflow.operators.parent_module
```